devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,579 @@
+	def dimmerEvent = createEvent([name: "level", value: cmd.value == 99 ? 100 : cmd.value, unit: "%"])
+	def result = [switchEvent, dimmerEvent]
+	if (switchEvent.isStateChange) {
+		result << response(["delay 1000", zwave.meterV3.meterGet(scale: 2).format()])
+	}
+	return result
+}
+
+def on() {
+	encapSequence([
+		zwave.basicV1.basicSet(value: 0xFF),
+		zwave.basicV1.basicGet()
+	], 1000)
+}
+
+def off() {
+	encapSequence([
+		zwave.basicV1.basicSet(value: 0x00),
+		zwave.basicV1.basicGet()
+	], 1000)
+}
+
+def setLevel(level) {
+	log.debug "setLevel"
+	if (level > 99) level = 99
+	encapSequence([
+		zwave.basicV1.basicSet(value: level),
+		zwave.switchMultilevelV1.switchMultilevelGet()
+	], 1000)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1 && cmd.scale == 0) {
+		map = [name: "energy", value: cmd.scaledMeterValue, unit: "kWh"]
+	} else if (cmd.meterType == 1 && cmd.scale == 2) {
+		map = [name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W"]
+	}
+	createEvent(map)
+}
+
+private getButtonLabel() {
+	[
+		"Up",
+		"Down",
+		"Configuration"
+	]
+}
+
+private void createChildButtonDevices() {
+	for (buttonNumber in 1..3) {
+		def child = addChildDevice("smartthings", "Child Button", "${device.deviceNetworkId}:${buttonNumber}", device.hub.id,
+			[
+				completedSetup: true,
+				label         : buttonLabel[buttonNumber - 1],
+				isComponent   : true,
+				componentName : "button$buttonNumber",
+				componentLabel: buttonLabel[buttonNumber - 1]
+			])
+
+		child.sendEvent(name: "supportedButtonValues", value: ["pushed"].encodeAsJSON(), displayed: false)
+		child.sendEvent(name: "numberOfButtons", value: 1, displayed: false)
+		child.sendEvent(name: "button", value: "pushed", data: [buttonNumber: 1], displayed: false)
+	}
+}
+
+def sendButtonEvent(gesture, buttonNumber) {
+	def event = createEvent([name: "button", value: gesture, data: [buttonNumber: buttonNumber], isStateChange: true])
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+	return createEvent([name: "button", value: gesture, data: [buttonNumber: buttonNumber], isStateChange: true, displayed: false])
+}
+
+def labelForGesture(gesture, attribute) {
+	if (attribute == 0) {
+		gesture;
+	} else {
+		def number = attribute - 1;
+		"${gesture}_${number}x";
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	log.info("CentralSceneNotification, keyAttributes=${cmd.keyAttributes}, sceneNumber=${cmd.sceneNumber}")
+	def singleClick = 0;
+	def multipleClicks = [3, 4, 5, 6]
+	def supportedAttributes = [singleClick] + multipleClicks
+	int attribute = cmd.keyAttributes
+	int scene = cmd.sceneNumber
+	if (scene == 1 && attribute in supportedAttributes) {
+		sendButtonEvent(labelForGesture("down", attribute), 2);
```

Comment:
The "down" and "up" in the capability don't refer to the directions of the buttons themselves, but the state of the button. So a "pushed" event could actually sometimes be a "down" event followed by an "up" event.

I think these should be replaced by "pushed".

Reply:
Done